### PR TITLE
RDCC-2326: Increase Mutation Coverage

### PR DESCRIPTION
**Pull request by AdilOoze**
### JIRA link (if applicable) ###

[RDCC-2326](https://tools.hmcts.net/jira/browse/RDCC-2326)

### Change description ###

Increasing Mutation Coverage >90%

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/service/CaseWorkerProfileConverter.java [hunk 1]
```diff
@@ -117,7 +117,7 @@ private boolean isSuspended(CaseWorkerProfile obj) {
         return caseWorkerLocationRequests;
     }
 
-    private Set<String> generateIdamRoles(String idamRoles) {
+    public Set<String> generateIdamRoles(String idamRoles) {
```
Comment (SujeethKumarAri) [inline src/main/java/uk/gov/hmcts/reform/cwrdapi/service/CaseWorkerProfileConverter.java:120]: Any reason for changing it to public?

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/service/CaseWorkerProfileConverter.java [hunk 2]
```diff
@@ -70,7 +70,7 @@
         return caseWorkersProfileCreationRequests;
     }
 
-    private boolean isSuspended(CaseWorkerProfile obj) {
+    public boolean isSuspended(CaseWorkerProfile obj) {
         return "Y".equals(obj.getSuspended());
```
[the remark this answers is not in the document]
Comment (AdilOoze) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/service/CaseWorkerProfileConverter.java:74]: Makes it easier to directly test the method and cover any mutations - can revert if necessary

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/service/CaseWorkerProfileConverter.java (hunk 1 above)
Comment (AdilOoze) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/service/CaseWorkerProfileConverter.java:120]: Makes it easier to directly test the method and cover any mutations - can revert if necessary

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/service/CaseWorkerProfileConverter.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (SujeethKumarAri) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/service/CaseWorkerProfileConverter.java:74]: Thanks for the explanation :) Actually, we should not change the access modifier for test. If we need to cover these private method then we need to cover using the actual public method that calls this method

[the remark this answers is not in the document]
Comment (AdilOoze) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/service/CaseWorkerProfileConverter.java:74]: Sure, I will revert :)

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/service/CaseWorkerProfileConverter.java (hunk 1 above)
Comment (AdilOoze) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/service/CaseWorkerProfileConverter.java]: Reverted here: https://github.com/hmcts/rd-caseworker-ref-api/pull/240/commits/b26a96500916bb6ad0b68f3869d4bf415045d440

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/service/CaseWorkerProfileConverter.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (AdilOoze) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/service/CaseWorkerProfileConverter.java]: Reverted here: https://github.com/hmcts/rd-caseworker-ref-api/pull/240/commits/b26a96500916bb6ad0b68f3869d4bf415045d440

File: build.gradle [hunk 3]
```diff
@@ -257,7 +257,7 @@ pitest {
   testSourceSets = [sourceSets.integrationTest, sourceSets.test]
   excludedClasses = [
     'uk.gov.hmcts.reform.cwrdapi.config.*',
-    'uk.gov.hmcts.reform.cwrdapi.controller.*',
+    'uk.gov.hmcts.reform.cwrdapi.controllers.*',
```
Comment (nitishsingla1234) [inline build.gradle:260]: Can we exclude the domain/model classes from pitest? I believe we have not written any logic in model classes(like equals/hashcode or any other method.)

Comment (AdilOoze) [reply build.gradle:260]: We can - however they are fully covered by unit tests now so would be a wasted effort

**Review by nitishsingla1234 — APPROVED** [review]

**Review by shreedhar-hmcts — APPROVED** [review]

**Review by SujeethKumarAri — APPROVED** [review]

**Pull request merged**